src/_avif.c

```diff
@@ -0,0 +1,944 @@
+    if (result != AVIF_RESULT_OK) {
+        PyErr_Format(
+            exc_type_for_avif_result(result),
+            "Setting IO memory failed: %s",
+            avifResultToString(result)
+        );
+        avifDecoderDestroy(decoder);
+        PyBuffer_Release(&buffer);
+        PyObject_Del(self);
+        return NULL;
+    }
+
+    result = avifDecoderParse(decoder);
+    if (result != AVIF_RESULT_OK) {
+        PyErr_Format(
+            exc_type_for_avif_result(result),
+            "Failed to decode image: %s",
+            avifResultToString(result)
+        );
+        avifDecoderDestroy(decoder);
+        PyBuffer_Release(&buffer);
+        PyObject_Del(self);
+        return NULL;
+    }
+
+    if (decoder->alphaPresent) {
+        self->mode = "RGBA";
+    } else {
+        self->mode = "RGB";
+    }
+
+    self->decoder = decoder;
+    self->buffer = buffer;
+
+    return (PyObject *)self;
+}
+
+PyObject *
+_decoder_dealloc(AvifDecoderObject *self) {
+    if (self->decoder) {
+        avifDecoderDestroy(self->decoder);
+    }
+    PyBuffer_Release(&self->buffer);
+    Py_RETURN_NONE;
+}
+
+PyObject *
+_decoder_get_info(AvifDecoderObject *self) {
+    avifDecoder *decoder = self->decoder;
+    avifImage *image = decoder->image;
+
+    PyObject *icc = NULL;
+    PyObject *exif = NULL;
+    PyObject *xmp = NULL;
+    PyObject *ret = NULL;
+
+    if (image->xmp.size) {
+        xmp = PyBytes_FromStringAndSize((const char *)image->xmp.data, image->xmp.size);
+    }
+
+    if (image->exif.size) {
+        exif =
+            PyBytes_FromStringAndSize((const char *)image->exif.data, image->exif.size);
+    }
+
+    if (image->icc.size) {
+        icc = PyBytes_FromStringAndSize((const char *)image->icc.data, image->icc.size);
+    }
+
+    ret = Py_BuildValue(
+        "IIIsSSIS",
+        image->width,
+        image->height,
+        decoder->imageCount,
+        self->mode,
+        NULL == icc ? Py_None : icc,
+        NULL == exif ? Py_None : exif,
+        irot_imir_to_exif_orientation(image),
+        NULL == xmp ? Py_None : xmp
+    );
+
+    Py_XDECREF(xmp);
+    Py_XDECREF(exif);
+    Py_XDECREF(icc);
+
+    return ret;
+}
+
+PyObject *
+_decoder_get_frame(AvifDecoderObject *self, PyObject *args) {
+    PyObject *bytes;
+    PyObject *ret;
+    Py_ssize_t size;
+    avifResult result;
+    avifRGBImage rgb;
+    avifDecoder *decoder;
+    avifImage *image;
+    uint32_t frame_index;
+
+    decoder = self->decoder;
+
+    if (!PyArg_ParseTuple(args, "I", &frame_index)) {
+        return NULL;
+    }
+
+    result = avifDecoderNthImage(decoder, frame_index);
+    if (result != AVIF_RESULT_OK) {
+        PyErr_Format(
+            exc_type_for_avif_result(result),
+            "Failed to decode frame %u: %s",
+            decoder->imageIndex + 1,
+            avifResultToString(result)
+        );
+        return NULL;
+    }
+
+    image = decoder->image;
+
+    memset(&rgb, 0, sizeof(rgb));
```

Comment:
Delete this `memset` call. The `avifRGBImageSetDefaults` call in the next line is the preferred way to initialize `rgb`.

Reply:
Thanks. That change has now been made.